concrete/controllers/single_page/dashboard/system/permissions/trusted_proxies.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Concrete\Controller\SinglePage\Dashboard\System\Permissions;
+
+use Concrete\Core\Page\Controller\DashboardPageController;
+use IPLib\Factory;
+use IPLib\Range\Pattern;
+
+class TrustedProxies extends DashboardPageController
+{
+    public function view()
+    {
+        $config = $this->app->make('config');
+        $trustedIPs = $config->get('concrete.security.trusted_proxies.ips');
+        if (!is_array($trustedIPs)) {
+            $trustedIPs = [];
+        }
+        $this->set('trustedIPs', $trustedIPs);
+    }
+
+    public function save()
+    {
+        if ($this->token->validate('ccm_trusted_proxies_save')) {
+            $post = $this->request->request;
+            $trustedIPs = [];
+            $invalidIPs = [];
+            $rawTrustedIPs = $post->get('trustedIPs');
+            if (is_string($rawTrustedIPs)) {
+                if (preg_match_all('/\S+/', $rawTrustedIPs, $matches)) {
+                    foreach ($matches[0] as $rawRange) {
+                        if (!in_array($rawRange, $invalidIPs, true)) {
+                            $range = Factory::rangeFromString($rawRange);
+                            if ($range === null || $range instanceof Pattern) {
```

Comment:
The cyclomatic complexity here is through the roof, this deserves to be split into a few methods